### ISSUE TITLE
Scaling Issues with Matte Layer

### ISSUE THREAD

**Issue by JamieJuliusSundaySky** [airbnb/lottie-android#416]
It seems that the presence of a matte layer causes incorrect scaling.

There was an issue with this in release 2.1.0 and now there seems to be a different issue in 2.1.2.

The attached zip contains a document and JSON that describe the behavior.

[Android Scaling Issues With Matte Layer.zip](https://github.com/airbnb/lottie-android/files/1192946/Android.Scaling.Issues.With.Matte.Layer.zip)

**Comment by gpeal** [comment]
@JamieJuliusSundaySky I can take another look at this. In the mean time, keep in mind that composition size in pixels in AE will be converted to dps on your phone. What I recommend doing is to set the composition to the size you want or slightly smaller even then use centerInside as the scale type. There is some performance overhead and funky calculations that have to be done if the animation has masks or mattes and is larger than the view canvas.

**Comment by gpeal** [comment]
@JamieJuliusSundaySky here is a little bit of background on the issue: https://github.com/airbnb/lottie-android/blob/master/lottie/src/main/java/com/airbnb/lottie/LottieDrawable.java#L303

**Comment by tobalr** [comment]
I just stumbled upon this issue as well. A temporary workaround is to calculate the scale for the animation by yourself 

```
 private float calculateCorrectLottieScale() {
        Display display = getWindowManager().getDefaultDisplay();
        Point size = new Point();
        display.getSize(size);
        double width = size.x;
        double height = size.y;

        //Calculate how the animation should be scaled to fit within screen
        float screenScale = (float) Math.min(width / AE_COMPOSITION_WIDTH, height / AE_COMPOSITION_HEIGHT);

        //Calculate scale to compencate for issue
        Resources r = getResources();
        float px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 1, r.getDisplayMetrics());
        float scale = 1f / px;
        return scale*screenScale;
    }
```

**Comment by gpeal** [comment]
I think this will be fixed bt #461

**Comment by gpeal** [comment]
@JamieJuliusSundaySky @tobalr v2.2.4 is now out
